### DISCUSSION TITLE
Need directions on handling the scenario: MsalClientException: Sdk canceled the auth flow

### DISCUSSION THREAD

**Discussion by ghost** [AzureAD/microsoft-authentication-library-for-android#1807] [Q&A]
We use the MSAL library for password less login using a phone number and OTP code sent to the phone. I'm running into a scenario we see an error.

1. User launches the app and starts the login flow
2. Enter the phone number to receive an OTP code.
3. Switches to the "Message" app to see the OTP (the app is still running, not visible to the user)
4. Instead of returning to the running app from recent apps (Depending on the Android phone model, this is achieved by a button or swipe up), **the User launches the app again by going to the list of apps and tapping on the app icon.** 
5. A new instance of the app(activity) is launched, and the MSAL auth library recognizes there was an ongoing authentication flow earlier and cancels it, and throws an error as it considers launching a new authentication request

`com.microsoft.identity.client.exception.MsalClientException: Sdk canceled the auth flow as the app launched a new interactive auth request.`

We used the '**SingleTop**' launch mode to open the existing activity if it exists, but that doesn't help. (**SingleInstance** doesn't help either)

Is there a way to resume the authentication flow from where it was left in this scenario? or any recommended way of handling the exception thrown by the MSAL auth library?

**Comment by negoe** [comment]
@slakshmanan-isn, For any GitHub issues we recommend posting your questions [here ](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues). Can you also please share your code implementation if it is open source or logs at least if possible for us to investigate deeper?

**Reply by ghost** [reply]
Thank you @negoe I posted it on the issues with the required details: https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/1819
